corehq/apps/smsbillables/models.py

```diff
@@ -353,3 +373,41 @@ def create(cls, message_log, api_response=None):
         billable.save()
 
         return billable
+
+    @classmethod
+    def _get_twilio_charges(cls, billable, message_log, backend_instance):
+        def _get_twilio_client(twilio_backend_id):
+            twilio_backend = SQLMobileBackend.get(twilio_backend_id)
+            account_sid = twilio_backend.account_sid
+            auth_token = twilio_backend.auth_token
+            return TwilioRestClient(account_sid, auth_token)
+
+        currency = None
+        try:
+            if message_log.backend_message_id:
+                twilio_message = _get_twilio_client(
+                    backend_instance
+                ).messages.get(message_log.backend_message_id)
```

Comment:
just twilio because we're pulling the fee from twilio itself and not applying a fee from our database. there's the possibility of a timing issue here since we'll create the billable task once we hand off the sms to twilio, but twilio might not process it by the time the billable task runs.  so before we get the fee api response from twilio we should check the message status from twilio and if it's any of the ones i listed above we should retry the billable task later.